index.js

```diff
@@ -37,6 +37,7 @@ function BinarySplit (matcher) {
       } else {
         if (offset >= buf.length) {
```

Comment:
Hmm, I'm also not sure whether this block is necessary there. @maxogden is it safe to remove or does it have a special meaning that's not obvious?